src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -37,6 +38,41 @@ public T WithHttpClientFactory(IMsalHttpClientFactory httpClientFactory)
             return (T)this;
         }
 
+        /// <summary>
+        /// Allows developers to configure their own valid authorities. A json string similar to https://aka.ms/aad-instance-discovery should be provided.
+        /// MSAL uses this information to: 
+        /// <list type="bullet">
+        /// <item>Call REST APIs on the environment specified in the preferred_network</item>
+        /// <item>Identify an environment under which to save tokens and accounts in the cache</item>
+        /// <item>Use the environment aliases to match tokens issued to other authorities</item>
+        /// </list>
+        /// For more details see https://aka.ms/msal-net-custom-instance-metadata
+        /// </summary>
+        /// <remarks>
+        /// Developers take responsibility for authority validation if they use this method. Should not be used when the authority is not know in advance. 
+        /// Has no effect on ADFS or B2C authorities, only for AAD authorities</remarks>
+        /// <param name="instanceDiscoveryJson"></param>
+        /// <returns></returns>
+        public T WithInstanceDicoveryMetadata(string instanceDiscoveryJson)
+        {
+            if (string.IsNullOrEmpty(instanceDiscoveryJson))
+                throw new ArgumentNullException(instanceDiscoveryJson);
```

Comment:
for now, for consistency with the entire rest of our codebase which follows this rule, can you put the {} on here and we'll start the mail thread up?